pkg/controller/render/render_controller.go

```diff
@@ -433,7 +433,20 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 		return fmt.Errorf("no MachineConfigs found matching selector %v", selector)
 	}
 
-	return ctrl.syncGeneratedMachineConfig(pool, mcs)
+	if err := ctrl.syncGeneratedMachineConfig(pool, mcs); err != nil {
+		return ctrl.syncFailingStatus(pool, err)
+	}
+
+	return nil
+}
+
+func (ctrl *Controller) syncFailingStatus(pool *mcfgv1.MachineConfigPool, err error) error {
+	sdegraded := mcfgv1.NewMachineConfigPoolCondition(mcfgv1.MachineConfigPoolDegraded, v1.ConditionTrue, fmt.Sprintf("Failed to render MCs for MCP %s: %v", pool.Name, err), "")
```

Comment:
Maybe just "config" and "pool" instead of "MC" and "MCP"?

Reply:
fixed thanks